javalib/src/main/scala/java/io/RandomAccessFile.scala

```diff
@@ -76,20 +76,39 @@ class RandomAccessFile private (file: File,
     in.readInt()
 
   override final def readLine(): String = {
-    if (getFilePointer() == length()) null
-    else {
+    // DataInputStream#readLine has been deprecated since JDK 1.1
+    // so implement RAF#readLine, rather than delegating.
+    var pos = getFilePointer()
+    var end = length() // standard practice: 1 past last valid byte.
+    if (pos >= end) {
+      null // JDK 8 specification requires null here.
+    } else {
       val builder = new StringBuilder
```

Comment:
This isn't specifically part of the PR but I think we should be using `java.lang.StringBuilder` here to remove the Scala lib dependency.

Reply:
Good suggestion, good eyes!  Thank you.  Review leads to better code all around.

Since that line is new code, I think that making the change you suggest does not
break the "unity of purpose" required of a PR.  

Easy enough to write to current practices and not introduce technical debt.

Other Scala lib dependencies may have crept in but I know of no easy way
to find them all at once.  At least this one is gone.  Others as they are found
(We will have to have a planned, but "Turn off Predef" bugfix week sometime).